### ISSUE TITLE
Add option to VS to migrate a project to the X# runtime

### ISSUE THREAD

**Issue by RobertvanderHulst** [X-Sharp/XSharpPublic#32]
Add a context option to the solution explorer tree to migrate a project (or all projects in a solution ?) to the X# runtime. This should at least do:

- replace Vulcan runtime DLLs with X# runtime DLLs
- check for hardcoded type names in the source code that refer to "Vulcan", like "Vulcan.Error", "VulcanRTFuncs" etc. Either fix these or mark them with a #warning

**Comment by VolkmarR** [comment]
Just an idea: Since this operation is executed only once per project, it might be overkill to put an menu item in popup menu of the the solution explorer. 

An alternative approach could be to create a "Project Converter Tool" (similar to the one DevExpress created https://docs.devexpress.com/ProjectConverter/2530/dialog-window) under Tools -> X#. 

This Project Converter lets you select a solution, project or folder and perform an operation. It this case, the operation is "migrate the project(s) to the X# runtime". 

Other operations could be
* Separate Winforms design code to designer file
* Convert from .net framework to .net x
* ...

**Comment by RobertvanderHulst** [comment]
Volkmar,
We have such a project / solution converter tool already. Tools / XSharp / Vulcan Exporter. At the moment this tool converts solutions with .vnproj files to a solution with .xsproj files
The work to change a xsproj from the Vulcan Runtime to the X# runtime is really a no brainer:
- removed old references
- add new references
And the work to integrate this in the IDE is already done. There is an entry in the VSCT file with Dynamic visibility. The command already checks for the existence of the Vulcan Runtime assemblies. When not, then the command is (stays) hidden on the context menu,
All we have to do is to add a search/replace table for assembly names, and replace assemblies.
In fact I might be able to do this this afternoon as an "easy snack".

**Comment by cpyrgas** [comment]
Can't find this option, where is it?

**Comment by RobertvanderHulst** [comment]
If you have a project that has Vulcan runtime DLLs then you can find it on the context menu of the project in the Solution Explorer.
Maybe I should also add it to the Project Menu ?

**Comment by cpyrgas** [comment]
Ah, it should have references to the vulcan runtime...I thought it would appear even without that, for checking hardcoded vulcan names, even after the references have been attached. The context menu is enough I think.

**Comment by cpyrgas** [comment]
It indeed works when there are vulcan references! But I see no warning or change for hardcoded vulcan names that you mentioned in the original report.

**Comment by RobertvanderHulst** [comment]
Chris,
You are right. I did not do that. But the first time you compile the compiler will mark these as error. So is that really necessary. 
What I can do is add a readme.txt to the project after the conversion that contains the next steps to take.
Do you think that would be sufficient ?

**Comment by cpyrgas** [comment]
Robert, yeah, that would be fine, maybe it's not even needed at all since as you say there will be compiler errors showing the problem anyway. I did not confirm this ticket as done only because you mentioned this extra feature in the opening text and I didn't see it, not saying it should definitely be done.

What might make more sense and would be very helpful when porting from vulcan would be to automatically update such vulcan references in the code to the equivalent X# ones, but that's probably too much work, prone to error and most importantly it will probably only affect just a handful remaining people that still have vulcan apps to be ported to X#...

So if you'd like close the issue even as it is now, it's probably already more than enough.
